_layouts/single.html

```diff
@@ -17,21 +17,30 @@
 <div id="main" role="main">
   {% include sidebar.html %}
 
-  <article class="page" itemscope itemtype="https://schema.org/CreativeWork">
+  <article class="page h-entry" itemscope itemtype="https://schema.org/CreativeWork">
     {% if page.title %}<meta itemprop="headline" content="{{ page.title | markdownify | strip_html | strip_newlines | escape_once }}">{% endif %}
     {% if page.excerpt %}<meta itemprop="description" content="{{ page.excerpt | markdownify | strip_html | strip_newlines | escape_once }}">{% endif %}
     {% if page.date %}<meta itemprop="datePublished" content="{{ page.date | date_to_xmlschema }}">{% endif %}
     {% if page.last_modified_at %}<meta itemprop="dateModified" content="{{ page.last_modified_at | date_to_xmlschema }}">{% endif %}
 
+  <ul class="hidden">
+    <li>These list items are microformat entries and are hidden from view.</li>
+    <li class="u-url">{{site.url}}{{page.url}}</li>
+    {% if site.author.name %}<li><a class="p-author h-card" href="{{ site.url }}"><img class="photo" src="{{ author.avatar | absolute_url }}">{{ site.author.name | markdownify | strip_html | strip_newlines | escape_once }}</a></li>{% endif %}
```

Comment:
Careful with your author variables. You start with `site.author` which will cause problems if someone has assigned an author to the post where it should [override the "site" author](https://mmistakes.github.io/minimal-mistakes/layout-author-override/). 

You're also mixing in a `{{ author }}` variable to do the avatar. They probably should all be `{{ author }}`. Take a look at how the [`assign` works in the author sidebar](https://github.com/mmistakes/minimal-mistakes/blob/master/_includes/author-profile.html#L1-L2) to see how I did it there. 

A conditional is needed for `author.avatar` since we can't guarantee someone set an image and wouldn't want an empty `<img>`.

Either way I question if the author microformat markup is even needed here as you've added it to the sidebar already. Will it fail without it? Just trying to get a feel for how DRY this can be since I'm not really thrilled with all the duplicate markup.

Reply:
Oh, wow... an excellent point about the mixing of author variables.  I will fix that with an update to the branch.

Regarding the author microdata markup in two places.  The markup in the sidebar is describing an [h-card](https://microformats.org/wiki/h-card).  In the microformats world, it seems like the site homepage should contain this markup to describe the controlling entity of the site.  The author markup in the "_single.html" file covers the stated author of the article on the page for an [h-entry](http://microformats.org/wiki/h-entry).  (And that is where I'm definitely misusing `{{ site.author }}`.

I'm not familiar with the mechanics of multi-author Jekyll blogs, but from what I can gather the `site` variables are appropriate for the sidebar (the "h-card" microformat covering the entire site) while the `author` variables are appropriate for the article itself (the "h-entry" microformat).